<launch>

    <!-- Controller Node -->
<!--     <node pkg="quad_control" name="plot_nodeasas" type="plot_experiment.py"  output="screen"/> -->

    <!-- Default Mission -->
    <param name="MissionDefaultIrisSimulator" type="str" value="IrisSimulatorTrajectoryTracking"/>

    <!-- Default Yaw Controller -->
    <param name="YawControllerDefault" type="str" value="SimpleTrackingYawController"/>

    <!-- Default Trajectory -->
    <param name="TrajectoryDefault" type="str" value="StayAtRest"/>
    <param name="trajectry_offset_x" type="double" value="0.0"/>
    <param name="trajectry_offset_y" type="double" value="0.0"/>
    <param name="trajectry_offset_z" type="double" value="0.0"/>


    <!-- Default Yaw Controller -->
    <param name="ControllerDefault" type="str" value="SimplePIDController"/>
    <param name="DIControllerDefault" type="str" value="ComponentWise3DDIC"/>


    <!-- Default Vector Thrust Controller -->
    <param name="VThrustControllerDefault" type="str" value="VThrustQuadrupleController"/>

    <!-- Default Quadruple Integrator Controller -->
    <param name="QuadrupleIntegratorController" type="str" value="LinearQuadrupleIntController"/>

    <!-- Default Controller for Load transportation by single aerial vehicle -->
    <param name="SingleLoadTransportationControllerDefault" type="str" value="SingleLoadTransportController"/>

    <!-- Controller Node -->
    <node pkg="quad_control" name="controller_Iris1" type="cycle_quad_control_mission.py" output="screen"/>

    <!-- Controller parameters -->
    <param name="gravity_ctr" value="9.81"/>
    <param name="mass_quad_ctr" value="1.442"/>
    <param name="Throttle_neutral_ctr" value="1484.0"/>  
    <param name="ACRO_RP_P_ctr" value="4.5"/> 
    <param name="ktt" value="20.0"/> 
    <param name="ktt2" value="30.0"/>
    <!--  SOME USEFUL RULES: kp = wn*wn  AND kv = 2 xsi wn--> 
    <param name="kv" value="1.0"/> 
    <param name="kp" value="1.4"/> 
    <param name="sigma_p" value="0.5"/> 
    <param name="sigma_v" value="0.5"/> 

    <param name="throttle_neutral" value="1450"/> 
    <param name="quadrotor_mass" value="1.442"/>

    <!-- load lifting: cable length in meters -->
    <param name="cable_length" value="0.6"/> 
    <!-- load lifting: load mass in kg -->
    <param name="load_mass" value="0.1"/>

    
</launch>